lib/OpenQA/WebAPI/Controller/API/V1/Worker.pm

```diff
@@ -158,16 +160,30 @@ sub create {
     my $instance = $self->param('instance');
     my $caps     = {};
 
-    $caps->{cpu_modelname} = $self->param('cpu_modelname');
-    $caps->{cpu_arch}      = $self->param('cpu_arch');
-    $caps->{cpu_opmode}    = $self->param('cpu_opmode');
-    $caps->{mem_max}       = $self->param('mem_max');
-    $caps->{worker_class}  = $self->param('worker_class');
+    $caps->{cpu_modelname}                = $self->param('cpu_modelname');
+    $caps->{cpu_arch}                     = $self->param('cpu_arch');
+    $caps->{cpu_opmode}                   = $self->param('cpu_opmode');
+    $caps->{mem_max}                      = $self->param('mem_max');
+    $caps->{worker_class}                 = $self->param('worker_class');
+    $caps->{websocket_api_version}        = $self->param('websocket_api_version');
+    $caps->{isotovideo_interface_version} = $self->param('isotovideo_interface_version');
+
+    my $id;
+    try {
+        $id = $self->_register($self->db, $host, $instance, $caps);
+    }
+    catch {
+        if (/Incompatible/) {
+            $self->render(status => 426, json => {error => $_});
+        }
+        else {
+            die $_;
```

Comment:
I don't think the webUI needs to die here either

Reply:
@foursixnine 

I only want to capture one specific exception. All the other exceptions make the server to return 500 error code. That's what we want.